assets/js/modules/tagmanager/datastore/settings.js

```diff
@@ -88,6 +88,11 @@ export async function submitChanges( { select, dispatch } ) {
 	// This action shouldn't be called if settings haven't changed,
 	// but this prevents errors in tests.
 	if ( select( STORE_NAME ).haveSettingsChanged() ) {
+		// Clear any errors that may have been generated while creating container.
+		await dispatch( STORE_NAME ).clearErrors( 'createContainer' );
+		// Clear any error (in state.error) that legacy code may have reported.
+		await dispatch( STORE_NAME ).clearError();
```

Comment:
This solves the bug on the issue.

Reply:
@kostyalmm would you please provide a bit more detail as to what specifically this is solving?.

Reply:
The problem mentioned in the issue seems to be fixed already before I took the issue. Can you confirm?
I was confused and made up another (unrealistic) flow altogether on my own 😓 . 
This change is not required, reverted it back.